src/runtime/patch/fs/patcher.js

```diff
@@ -0,0 +1,99 @@
+import * as realFs from 'fs';
+import * as util from 'util';
+import { createPatchedFs } from './fs';
+
+const MemberType = {
+  Constructor: 0,
+  Property: 1,
+  PromisesApi: 2,
+};
+
+const metadata = {
+  StatWatcher: MemberType.Constructor,
+  FSWatcher: MemberType.Constructor,
+  ReadStream: MemberType.Constructor,
+  WriteStream: MemberType.Constructor,
+  ReadFileStream: MemberType.Constructor,
+  WriteFileStream: MemberType.Constructor,
+  Stats: MemberType.Constructor,
+
+  constants: MemberType.Property,
+  F_OK: MemberType.Property,
+  R_OK: MemberType.Property,
+  W_OK: MemberType.Property,
+  X_OK: MemberType.Property,
+
+  promises: MemberType.PromisesApi,
+};
+
+function applyPatchFs(patchedFs, originalFs) {
+  // create a backup before modification
+  const backupFs = { ...originalFs };
+
+  // iterate over the filesystem and patch members
+  for (const member of Object.getOwnPropertyNames(originalFs)) {
+    if (!patchedFs[member] || typeof patchedFs[member] !== typeof originalFs[member]) {
```

Comment:
**question** Should we instead be throwing when this happens? If I understand correctly, additions to the Node.js `fs` API would go silently unpatched.

Reply:
There are a lot of methods we are not patching because they are write only and the static-fs is read only like: `renameSync`, `rmdirSync` etc. In order to proceed with the warnings I would need to keep a list of things to ignore and to not warning about it. I think unless we discovered the `fs` api changes too much frequently we can keep it like that for now